### PR TITLE
fix: switching to alloy for blocks

### PR DESCRIPTION
**Pull request by Freyskeyd**
# Description

This PR switches from ethers to alloy for listening on Blocks.

It uses a backoff mechanisms in case of network failure

## PR Checklist:

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added or updated tests that comprehensively prove my change is effective or that my feature works

### CODE REVIEW

File: crates/agglayer-clock/src/block.rs [hunk 1]
```diff
@@ -4,19 +4,36 @@ use std::{
...
 use tokio_util::sync::CancellationToken;
 use tracing::{debug, error, info, trace, warn};
 
 use crate::{Clock, ClockRef, Error, Event, BROADCAST_CHANNEL_SIZE};
 
+type BlockProvider = FillProvider<
+    JoinFill<
+        Identity,
+        JoinFill<GasFiller, JoinFill<BlobGasFiller, JoinFill<NonceFiller, ChainIdFiller>>>,
+    >,
```
Comment (iljakuklic) [inline crates/agglayer-clock/src/block.rs:31]: Nitpick:

```suggestion
    JoinFill<
        Identity,
        Ethereum::RecommendedFillers,
    >,
```

(probably needs to import / qualify with the `RecommendedFillers` trait, will not work as suggested)

File: crates/agglayer-clock/src/block.rs [hunk 2]
```diff
@@ -260,21 +315,59 @@ where
...
+                Ok(self.0.try_reconnect().await?)
+            },
+        )
+        .await
+    }
+}
+
 #[cfg(test)]
 mod tests {
-    use std::{num::NonZeroU64, time::Duration};
```
Comment (iljakuklic) [inline crates/agglayer-clock/src/block.rs:265]: Nitpick: Imho there's now enough tests to deserve a separate file.

**Review by iljakuklic — APPROVED** [review]
+1, no important comments, just some nits.

File: crates/agglayer-clock/src/block.rs (hunk 1 above)
Comment (Freyskeyd) [reply crates/agglayer-clock/src/block.rs:31]: I tried to use that format earlier, but it doesn't reduce the boilerplate as I needed to define `Ethereum as ...`

**Pull request merged**